### PR TITLE
drop old coordinates 'nlat' 'nlon' if present, rename z_w to z_w_top …

### PR DESCRIPTION
**Pull request by ALDepp**
…if both present

'nlat' and 'nlon' are old coordinates and should not be included in the new ds.
if 'z_w_top' and 'z_w' are both present, drop 'z_w' and rename to 'z_w_top', to facilitate, e.g., multiplication with variables on those levels (the depths of 'z_w' and 'z_w_top' are identical

### CODE REVIEW

File: pop_tools/xgcm_util.py [hunk 1]
```diff
@@ -201,5 +201,11 @@ def to_xgcm_grid_dataset(ds, **kwargs):
             """to_xgcm_grid_dataset() function requires the `xgcm` package. \nYou can install it via PyPI or Conda"""
         )
     ds_new = relabel_pop_dims(ds)
+    if 'nlat' in ds_new.dims:
+        ds_new = ds_new.drop('nlat')
+    if 'nlon' in ds_new.dims:
+        ds_new = ds_new.drop('nlon')
+    if 'z_w_top' and 'z_w' in ds_new.dims:
+        ds_new = ds_new.drop('z_w_top').rename({'z_w': 'z_w_top'})
```
Comment (andersy005) [inline pop_tools/xgcm_util.py]: Would moving these lines into `relabel_pop_dims()` function (right before the return statement) cause any problem?

https://github.com/NCAR/pop-tools/blob/b76f5c1e5536ad592a825743a2d27bad2f485b54/pop_tools/xgcm_util.py#L100-L102

**Comment by andersy005** [comment]
@ALDepp, the code-style issues can be addressed by installing `pre-commit` on your machine. The `pre-commit` tool will (automatically) run pre-commit hooks every time you run `git commit -m ...`, and will format your code before committing/pushing it to GitHub. 

```
$ conda install -c conda-forge pre-commit
$ pre-commit install # Run this from the main git directory (pop-tools)
```

File: pop_tools/xgcm_util.py (hunk 1 above)
Comment (ALDepp) [reply pop_tools/xgcm_util.py]: I think that should work as well.

Comment (dcherian) [reply pop_tools/xgcm_util.py]: Does this work? 

If `relabel_pop_dims` does its job, `nlat` and `nlon` should not be in `ds_new.dims` but will be in `ds_new.coords`. In that case you should use `.drop_vars` (does the same thing but is clearer about what's being dropped)

Comment (ALDepp) [reply pop_tools/xgcm_util.py]: so 
if 'nlat' in ds_new.coords:
    ds_new = ds_new.drop_vars('nlat')

and so on?

Comment (dcherian) [reply pop_tools/xgcm_util.py]: I think so.

We should add a test with `assert 'nlon' not in ds_new.dims` and so on.

Comment (ALDepp) [reply pop_tools/xgcm_util.py]: I changed ```drop``` to ```drop_vars``` and added a line with ```assert```, not sure if that is what you had in mind though.

Comment (dcherian) [reply pop_tools/xgcm_util.py]: Almost. Let's put the `assert` statement in this test:
https://github.com/NCAR/pop-tools/blob/b76f5c1e5536ad592a825743a2d27bad2f485b54/tests/test_xgcm_util.py#L14-L23

We should also `assert z_w not in ds_new`

You can run the tests by (after conda installing `pytest`)
```
cd pop-tools/tests
pytest
```

File: pop_tools/xgcm_util.py [hunk 2]
```diff
@@ -98,6 +98,13 @@ def relabel_pop_dims(ds):
             else:
                 dims = new_spatial_dims
             ds_new[vname] = xr.Variable(dims, da.data, da.attrs, da.encoding, fastpath=True)
+    old_coords = ['nlat', 'nlon']
+    for coord in old_coords:
+        if coord in ds_new.coords:
+            ds_new = ds_new.drop_vars(coord)
+        assert coord not in ds_new.dims
+    if 'z_w_top' and 'z_w' in ds_new.dims:
```
Comment (andersy005) [inline pop_tools/xgcm_util.py]: if `'z_w_top' and 'z_w' in ds_new.dims` --> `if 'z_w_top' in ds_new.dims and 'z_w' in ds_new.dims`:

File: tests/test_xgcm_util.py [hunk 3]
```diff
@@ -21,6 +21,9 @@ def test_to_xgcm_grid_dataset(file):
     for coord in new_spatial_coords:
         assert coord in ds_new.coords
         assert coord not in ds.coords
+    old_spatial_coords = set(['nlat', 'nlon', 'z_w'])
```
Comment (dcherian) [inline tests/test_xgcm_util.py]:
```suggestion
    old_spatial_coords = ['nlat', 'nlon', 'z_w']
```

**Comment by dcherian** [comment]
Anna's first PR!

**Pull request merged**